src/CacheManager.Core/BaseCacheManager.GetOrAdd.cs

```diff
@@ -20,7 +20,7 @@ public TCacheValue GetOrAdd(string key, Func<string, TCacheValue> valueFactory)
             NotNullOrWhiteSpace(key, nameof(key));
             NotNull(valueFactory, nameof(valueFactory));
 
-            return GetOrAddInternal(key, null, (k, r) => valueFactory(k));
+            return GetOrAddInternal(key, null, (k, r) => new CacheItem<TCacheValue>(k, valueFactory(k))).Value;
```

Comment:
AFAICT, there's no way the `CacheItem` returned from `GetOrAddInternal()` could ever be null? If a function generating null is passed into `GetOrAddInternal()` it will explicitly throw inside of it.